src/sentry/tasks/relay.py

```diff
@@ -254,3 +273,46 @@ def schedule_invalidate_project_config(
     projectconfig_debounce_cache.invalidation.debounce(
         public_key=public_key, project_id=project_id, organization_id=organization_id
     )
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.check_build_project_config",
+    queue="relay_config_check",
+)
+def check_build_project_config(public_key=None, **kwargs):
+    """Temporary task to verify a build task produces a cache entry
+
+    A build task should result in a cache entry being written.  Sometimes it seems this
+    doesn't work.  This task checks for cache entries and logs data if they are missing.  It
+    should be scheduled together with the build task itself.
+
+    This task needs to be scheduled in the future together with the original task.
+    """
+    from sentry.models import ProjectKey
+
+    metrics.incr(
+        "relay.projectconfig_cache.run",
+        tags={"task": "check"},
+        sample_rate=1,
+    )
+    if not public_key:
+        raise TypeError("public key must be present")
+    cfg = projectconfig_cache.get(public_key)
+    if not cfg:
+        metrics.incr("relay.projectconfig_cache.check_task.missing")
+
+        # Purposefully not catching exceptions for missing items from the DB, the task will
+        # fail and sentry will capture them.
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_extra("PublicKey", public_key)
+        key = ProjectKey.objects.get(public_key=public_key)
+        project_id = key.project.id
+        org_id = key.project.organization.id
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_extra("Project.id", project_id)
+            scope.set_extra("Organization.id", org_id)
+
+            sentry_sdk.capture_message("PublicKey not found in cache", level="warning")
+
+    else:
+        metrics.incr("relay.projectconfig_cache.check_task.found")
```

Comment:
Do we want to add sampling to this metric? Same for the other branch of the `if`. If no, I don't see the reason set max sampling to the `relay.projectconfig_cache.run` metric at the beginning of this task.

Reply:
yes, let's not sample while we're looking for this.  good catch